src/Sylius/Behat/Context/Api/Shop/CheckoutContext.php

```diff
@@ -553,6 +553,35 @@ public function iShouldBeOnTheCheckoutCompleteStep(): void
         );
     }
 
+    /**
+     * @Then I should be informed with :paymentMethod payment method instructions
+     */
+    public function iShouldBeInformedWithPaymentMethodInstructions(PaymentMethodInterface $paymentMethod): void
+    {
+        $response = $this->client->getLastResponse();
+        $payments = $this->responseChecker->getValue($response, 'payments');
+
+        Assert::notEmpty($payments, 'No payments found in response.');
+        $paymentMethodIri = $this->iriConverter->getIriFromItem($paymentMethod);
+        foreach ($payments as $payment) {
+            if ($payment['method'] === $paymentMethodIri) {
```

Comment:
```suggestion
            if ($payment['method'] !== $paymentMethodIri) {
                continue;
            }
```
Would make it easier to read.